docs/migration.md

```diff
@@ -0,0 +1,301 @@
+# Migration Guide
+
+This guide documents all of the changes between API Elements 0.6 and API
+Elements 1.0.
+
+## JSON Serialisation
+
+In prior versions of API Element, an element could be serialised in JSON as a
+JSON type, or an API Element.
```

Comment:
```
In prior versions of API Elements, an Element (i.e. a type) could be occasionally serialized as a value. This behaviour has since been dropped, so that Elements are always serialized in full form (i.e. as a type, not as a value). For example: ...
```
What about something like this?

Reply:
I think that's much better, copied this into da2b009.